server/gitwork.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+const TIME_INTERVAL = 6 * time.Hour
+
+type GitConfig struct {
+	RepositoryName string
+	RepositoryURL  string
+	RemoteName     string
+	LocalCommit    string
+	RemoteCommit   string
+}
+
+// Trigger is reposible for setting off the go routine for git-op
+func Trigger() {
+
+	object := GitConfig{
+		RepositoryName: os.Getenv("GOPATH") + "/src/github.com/litmuschaos/chaos-charts/",
+		RepositoryURL:  "https://github.com/litmuschaos/chaos-charts",
+		LocalCommit:    "",
+		RemoteCommit:   "",
+		RemoteName:     "origin",
+	}
+	for true {
+		object.GitOpDriver()
+		time.Sleep(TIME_INTERVAL)
+	}
+}
+
+// CheckERROR is used to handle errors thrown by any git-op called
+func CheckERROR(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// CheckIfRepositoryExists checks for the existence of this past existence of this repository
+func (object GitConfig) CheckIfRepositoryExists() (bool, error) {
```

Comment:
```
func (object GitConfig) CheckIfRepositoryExists() (bool, error) {
	_, err := os.Stat(object.RepositoryName)
	if err != nil {
		return false, err
	}
	if os.IsNotExist(err) {
		return false, nil
	}
	return true, nil
}
```

Reply:
i beg to disagree @chandankumar4 the code is correct and works well in all cases.
as for the code suggested by you, the code prints :
```
ERRO[0000] Error while checking repo exists, err: stat /home/daitya/go/src/github.com/litmuschaos/chaos-charts/: no such file or directory 
```

Reply:
updated the snippet by @chandankumar4 .. 
It's valid now.
cc: @aditya109